crates/biome_js_analyze/src/analyzers/nursery/no_unused_private_class_members.rs

```diff
@@ -0,0 +1,299 @@
+        .collect()
+}
+
+fn is_private(syntax: &SyntaxNode<JsLanguage>) -> bool {
+    syntax.descendants().any(|node| {
+        let is_es_private = JsPrivateClassMemberName::cast_ref(&node).is_some();
+        let is_ts_private = TsAccessibilityModifier::cast_ref(&node)
+            .is_some_and(|accessibility| accessibility.is_private());
+
+        is_es_private || is_ts_private
+    })
+}
+
+/// Check whether the js name is in a write only assignment.
+fn is_write_only(js_name: &AnyJsName) -> bool {
+    let grand_parent = js_name.syntax().grand_parent();
+    let is_assignment_expression = grand_parent
+        .kind()
+        .is_some_and(|kind| matches!(kind, JsSyntaxKind::JS_ASSIGNMENT_EXPRESSION));
+
+    // Skip every non assignment expression
+    if !is_assignment_expression {
+        return false;
+    }
+
+    if let Some(grand_parent) = grand_parent {
+        if let Some(parent) = js_name.syntax().parent() {
+            if let Some(assignment_expression) = JsAssignmentExpression::cast(grand_parent) {
+                let left = assignment_expression.left().ok();
+
+                if left.is_some_and(|left| !is_node_equal(left.syntax(), &parent)) {
+                    return false;
+                }
+
+                if !matches!(
+                    assignment_expression.operator(),
+                    Ok(JsAssignmentOperator::Assign)
+                ) {
+                    let kind = assignment_expression.syntax().parent().kind();
+                    return kind
+                        .is_some_and(|kind| matches!(kind, JsSyntaxKind::JS_EXPRESSION_STATEMENT));
+                }
+            }
+        }
+    }
+
+    true
+}
+
+fn is_update_expression(js_name: &AnyJsName) -> bool {
+    let grand_parent = js_name.syntax().grand_parent();
+
+    grand_parent.kind().is_some_and(|kind| {
+        matches!(
+            kind,
+            JsSyntaxKind::JS_POST_UPDATE_EXPRESSION | JsSyntaxKind::JS_PRE_UPDATE_EXPRESSION
+        )
+    })
+}
+
+fn get_member_text_token(member: &AnyJsClassMember) -> Option<TokenText> {
+    match member {
+        AnyJsClassMember::JsGetterClassMember(member) => member.name().ok()?.name(),
+        AnyJsClassMember::JsMethodClassMember(member) => member.name().ok()?.name(),
+        AnyJsClassMember::JsPropertyClassMember(member) => member.name().ok()?.name(),
+        AnyJsClassMember::JsSetterClassMember(member) => member.name().ok()?.name(),
+        _ => None,
+    }
+}
+
+fn get_member_range(member: &AnyJsClassMember) -> Option<TextRange> {
+    match member {
+        AnyJsClassMember::JsGetterClassMember(member) => Some(member.name().ok()?.range()),
+        AnyJsClassMember::JsMethodClassMember(member) => Some(member.name().ok()?.range()),
+        AnyJsClassMember::JsPropertyClassMember(member) => Some(member.name().ok()?.range()),
+        AnyJsClassMember::JsSetterClassMember(member) => Some(member.name().ok()?.range()),
+        _ => None,
+    }
+}
```

Comment:
These codes look pretty similar.
I suggest writing a method `get_member_name` that returns the token (the return type could be `Option<JsSyntaxToken>`).
At usage site you can call `name()` and `range()`.